storage/Devices/DeviceImpl.cc

```diff
@@ -283,7 +290,35 @@ namespace storage
     void
     Device::Impl::do_delete() const
     {
-	throw std::logic_error("stub function called");
+	ST_THROW(LogicException("stub function called"));
```

Comment:
How about declaring those methods as pure virtual so the compiler can generate code to catch this kind of thing?